### PR TITLE
wip: inappchat  component for virtual conference

### PR DESCRIPTION
**Pull request by sidmohanty11**
https://user-images.githubusercontent.com/73601258/153422109-ac96ef88-39c6-42b3-ab9f-30c3ff9ae13c.mp4

#38

### CODE REVIEW

**Comment by sidmohanty11** [comment]
## Fetching Messages And Sending Messages

https://user-images.githubusercontent.com/73601258/153489088-371d8a78-6552-4b7a-ac13-e9084bb0b45e.mp4

File: app/components/clientsideonly/fuselage.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import dynamic from "next/dynamic";
```
Comment (Sing-Li) [inline app/components/clientsideonly/fuselage.js:1]: Let's move this to `inappchat/lib/fuselage.js`

File: app/components/rocketchat.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import { useState } from "react";
```
Comment (Sing-Li) [inline app/components/rocketchat.js:1]: lets move this to `inappchat/inappchat.js`

File: app/lib/rocketchatapi.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// using env variables for now, should be replaced with cookies!
```
Comment (Sing-Li) [inline app/lib/rocketchatapi.js:1]: put this under the component's own `lib` directory - not the global one

File: app/package-lock.json [hunk 4]
```diff
@@ -8,8 +8,14 @@
       "name": "rc4community",
       "version": "0.3.0",
       "dependencies": {
+        "@rocket.chat/fuselage": "^0.31.3",
```
Comment (Sing-Li) [inline app/package-lock.json]: DO NOT commit this file --  get it into your `.gitignore`

File: app/styles/RocketChat.module.css [hunk 5]
```diff
@@ -0,0 +1,24 @@
+.sidechat {
```
Comment (Sing-Li) [inline app/styles/RocketChat.module.css:1]: change the name  ---   `Inappchat.module.css`

File: app/package.json [hunk 6]
```diff
@@ -9,6 +9,10 @@
     "lint": "next lint"
```
Comment (Sing-Li) [inline app/package.json:9]: I wonder if we can move these down to the componenet level -- making each one of our components an indepdendently installable npm module (optionally).

**Review by Sing-Li — COMMENTED** [review]
First review comments.

**Comment by Sing-Li** [comment]
> rocketchatcomponent1.mp4 
> #38

This is terrible UX  fwiw.   When there is no chat - we want the video to be in "full size" maxed out on the frame without sacrificing aspect.   The "tab" or "button" or "arrow" to activate chat should be up at the top right - as small as reasonable -- and be transparent if placed over the playing video.  On mobile, there should be no "tab"/"button"/"arrow" at all, but just a "swipe right" gesture.
In no case should be there be "jumbo white space" and "jumbo blue button" that hogs real estate for no purpose at all.

Please confer with irfan on the "design" for next iteration.

**Comment by sidmohanty11** [comment]
True, I've just placed them as placeholders for now.Will be working on them soon

File: app/components/clientsideonly/fuselage.js (hunk 1 above)
Comment (sidmohanty11) [reply app/components/clientsideonly/fuselage.js]: done

File: app/components/rocketchat.js (hunk 2 above)
Comment (sidmohanty11) [reply app/components/rocketchat.js]: done

File: app/lib/rocketchatapi.js (hunk 3 above)
Comment (sidmohanty11) [reply app/lib/rocketchatapi.js]: done

File: app/package-lock.json (hunk 4 above)
Comment (sidmohanty11) [reply app/package-lock.json]: added it to .gitignore and deleted it from commit

File: app/package.json (hunk 6 above)
Comment (sidmohanty11) [reply app/package.json:9]: I guess we can achieve this by making a monorepo and optionally dealing with components as we need. I will talk with Rohan sir about this and let you know

**Comment by Sing-Li** [comment]
@sidmohanty11   does the current code include the emoji support you have shown us today?

I'd like to marge it down ASAP so @demonicirfan  can add the animations.

Let me know.   It would be good if the cookie implementation can be added as well (but not necessarily at this time).

**Comment by sidmohanty11** [comment]
Just give me one more day so I can try my best to figure out if we can use the cookies and clean some things up

Thank you!

**Comment by sidmohanty11** [comment]
@Sing-Li , I have added the cookies part here (full-validation from browser cookies)

https://user-images.githubusercontent.com/73601258/154857956-f4be68f3-b789-4616-84bc-08c8cb600dd0.mp4

But we are facing a major roadblock, i.e, with the previous approach (with CSR approach) after a certain interval the requests are being revalidated which as I have discussed with Rohan sir and he agreed that that might not be the best approach. So it is still to be figured out. :(

**Comment by Sing-Li** [comment]
Cool.  @sidmohanty11    Let's go with the current way of  handling  (hard coded ENVIRONMENT VAR) for now.   Is it OK for me to merge this down now?

**Comment by sidmohanty11** [comment]
@Sing-Li I can work on that on a separate PR, and as UI needs a lot of work I think @demonicirfan can take it from here

**Pull request merged**